Allow setting texture data from 1x to fix some textures resetting randomly

riperiperi opened the pull request:
Expected targets:

- Deltarune Chapter 1+2
- Crash Team Racing
- Those new pokemon games idk (reset scale after battle/menu?)

This won’t help all games- for example SMTV issues. This might also be an aggressive solution, so it would be good to test if this has negative effects in other games.

I won’t be able to complete this for a bit so testing would help me make a decision when I can.

saldabain commented:
I've noticed that the screen is totally black while a pokemon is evolving in bdsp. Not sure if it is specific to this PR, but this is the one I use.

Nothing noteworthy appears in the logs.

Leischii commented:
I can confirm saldabain's comment. With this Pr, evolution and egg hatching animations are completly black. Not present with master or ldn.

gdkchan commented:
Does that happen while in native resolution, scaled or both?

saldabain commented:
> Does that happen while in native resolution, scaled or both?

I've only tried it out in a scaled resolution. (Standard, not custom)

Niwu34 commented:
Can easily be reproduced in the Pokémon Starter selection as well
It will show something if you start the game on Native res,
it will show a black screen if you choose to go into the game with 4x res

riperiperi commented:
I've pushed a fix for the screen going black. It seems that several scenes use MSAA targets (which were not allowed to scale), and opengl was having trouble blitting from an MSAA target (1x) to a screen bound target (2x). Now, MSAA targets are allowed to scale, and 2d copies force scale to propagate so they both match.

riperiperi commented:
Note: right now, msaa textures cannot retain their data through a scale change.

GamerzHell9137 commented:
With the new changes RF5 is upscaling properly now!

PR 
![image](https://user-images.githubusercontent.com/2002038/142968027-a567d15b-ce63-4c07-8160-1a6994e68d63.png)

Master
![image](https://user-images.githubusercontent.com/2002038/142968047-4be2b709-cfd4-4ed6-8794-f07faa7d73d6.png)

Cloratus commented:
How I put that fix in my game? it is a offline build?

poudink commented:
the builds are available in the first reply of this thread (the one from the github-actions bot). you can also build from source yourself.

0rb1t5l commented:
Sometimes Caligula Effect 2 renders graphics incorrectly like this after a scene change, using 2x scaling and auto anisotropic filtering, never had this happen on master. Using ryujinx-Release-1.0.0+e99bd25-win_x64 on 1060 GPU.
![ryujinx_capture_2021-11-28_12-43-36](https://user-images.githubusercontent.com/95211784/143832065-99e3134c-6754-4397-92a5-854426d3f115.png)

ActualMandM commented:
Sonic Colors: Ultimate now displays 3D when upscaling with the PR, even though it still shows it at 1x scale

PR:
![2021-12-03_18-30-01_Ryujinx](https://user-images.githubusercontent.com/15317421/144693766-109bb81f-5ce4-419d-8d99-d67c0b75387b.png)

Master:
![2021-12-03_18-29-08_Ryujinx](https://user-images.githubusercontent.com/15317421/144693754-c8d79d4a-4370-4854-a1f2-4e95faa92cf0.png)

Niwu34 commented:
Fixes the reset of the res scaler after each battle in My Hero One's Justice 2

0rb1t5l commented:
This fixes reset scale for SMT V as well.

ActualMandM commented:
Fixes Cruis'n Blast upscaling

Master:
![master](https://user-images.githubusercontent.com/15317421/148029821-3a4e4e81-5584-469f-830d-93822654974d.png)

PR:
![pr](https://user-images.githubusercontent.com/15317421/148029900-ff646273-0b6d-4ac1-9c97-00e2cae46da0.png)

GamerzHell9137 commented:
Monopoly Madness is properly upscaling too now 
![image](https://user-images.githubusercontent.com/2002038/149101726-cffefb22-5664-4723-b90f-a36041ad862e.png)

rasmysamy commented:
I have built and tested the latest commit, I can confirm that the evolutions, etc... work well on BDSP for me. Thank you very much for this fix, it's a huge improvement and I hope it can be merged soon!

ActualMandM commented:
Fixes *The Stanley Parable Ultra Deluxe* upscaling

Master:
![Master](https://user-images.githubusercontent.com/15317421/165878450-d9b08ec5-5211-4db9-8698-150697a50cbe.png)

PR:
![Pull Request](https://user-images.githubusercontent.com/15317421/165878456-e48cb117-618b-4b94-8f6f-57d42c840782.png)

CrusadingNinja commented:
Download links are expired

riperiperi commented:
I did find a case that triggers it constantly - The NSO N64 emulator. However... I don't think this should be happening, even if it is taking a slower path by upscaling the manually set data.

![image](https://user-images.githubusercontent.com/6294155/213810914-7c092937-301d-46fa-b8d1-c4bcdb663e3b.png)

riperiperi commented:
I've fixed the bug above and also added some code to automatically blacklist after too many writes. My theory for this happening in Unity games is that it triggers when a render target is prepared for new usage, perhaps by the guest clearing its data. Deltarune might be similar, as the menu select for that game essentially switches between different games. Hopefully the current tuning should be good enough to keep these games scaling.

gdkchan reviewed (approved):
lgtm, thanks.
